app/src/main/java/org/wildstang/wildrank/androidv2/activities/SyncActivity.java

```diff
@@ -50,10 +60,37 @@ public void onCreate(Bundle savedInstanceState) {
     }
 
     private void beginSync() {
-        if (!SyncUtilities.isFlashDriveConnected()) {
-            showDriveNotMountedWarning();
+        if (SyncUtilities.useSAF()) {
+            SyncUtilities.openFileChooser(this);
```

Comment:
To address the question about KitKat vs. Lollipop:

In theory, the SAF code DOES work on KitKat, but I don't have a KitKat device to test on, so I left KitKat on the old code-path. (I have an Android 5.0.1 tablet and an Android 7.1.1 phone, so I can test on Lollipop and Nougat).